### ISSUE TITLE
Deprecated gradle features

### ISSUE THREAD

**Issue by fpj** [pravega/pravega#3352]
**Problem description**
I'm observing the following warning:

```
Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.
See https://docs.gradle.org/4.5/userguide/command_line_interface.html#sec:command_line_warnings
```

when running:

```
./gradlew clean assemble publish
```

**Problem location**
Gradle build.

**Suggestions for an improvement**
Locate the deprecated features that this is referring to and update accordingly.

**Comment by fpj** [comment]
Here is one warning:

```
Using TaskInputs.file() with something that doesn't resolve to a File object has been deprecated. This is scheduled to be removed in Gradle 5.0. Use TaskInputs.files() instead.
```

which is being triggered during the execution of this task:

```
> Task :shared:controller-api:generateProto
```

and is due to this issue:

https://github.com/google/protobuf-gradle-plugin/issues/227

I have made the change locally and confirmed that the upgrade fixes the warning. I'm uneasy about upgrading a dependency in `0.5` at this point, so I'm move it to `0.6` and raise a PR to upgrade the dependency.

**Comment by RaulGracia** [comment]
Closed via #4001.
